test/src/org/labkey/test/tests/targetedms/TargetedMSChromatogramOptimizationTest.java

```diff
@@ -0,0 +1,140 @@
+        checker().verifyEquals("Invalid number of rows in transition optimization", getServerTableRowCount("TransitionOptimization"),
+                sizeOfTable(downloadedClibFile, "MoleculeTransitionOptimization"));
+        checker().verifyEquals("Invalid number of rows in Molecule", getServerTableRowCount("Molecule"),
+                sizeOfTable(downloadedClibFile, "Molecule"));
+        checker().verifyEquals("Invalid number of rows in MoleculePrecursor", getServerTableRowCount("MoleculePrecursor"),
+                sizeOfTable(downloadedClibFile, "MoleculePrecursor"));
+
+
+//        checker().verifyEquals("Invalid number of rows in MoleculePrecursorRetentionTime", getServerTableRowCount("MoleculePrecursorRetentionTime"),
+//                sizeOfTable(downloadedClibFile, "MoleculePrecursorRetentionTime"));
+//        checker().verifyEquals("Invalid number of rows in MoleculeList", getServerTableRowCount("MoleculeList"),
+//                sizeOfTable(downloadedClibFile, "MoleculeList"));
+
+
+    }
+
+    private int getServerTableRowCount(String tableName)
+    {
+        goToSchemaBrowser();
+        DataRegionTable table = viewQueryData("targetedms", tableName);
+        table.rowSelector().showAll();
+        try
+        {
+            table.goToView("LibraryMolecules");
+        }
+        catch (NoSuchElementException e)
+        {
+            return table.getDataRowCount(); // Default view count.
+        }
+
+        return table.getDataRowCount(); // Returns Library Molecules view count.
+    }
+
+    private int sizeOfTable(File clibFile, String name) throws SQLException
+    {
+        int cnt = 0;
+        @SuppressWarnings("SqlResolve")
+        String sql = "SELECT * FROM " + name;
+        try (Connection conn = ConnectionSource.getConnection(clibFile.getAbsolutePath()))
+        {
+            ResultSet rs = conn.createStatement().executeQuery(sql);
+            while (rs.next())
+                cnt++;
+        }
+        return cnt;
+    }
+
+    private boolean columnExists(File clibFile, String tableName, String columnName) throws SQLException
+    {
+        try (Connection conn = ConnectionSource.getConnection(clibFile.getAbsolutePath()))
+        {
+            DatabaseMetaData md = conn.getMetaData();
+            ResultSet rs = md.getColumns(null, null, tableName, columnName);
+            if (rs.next())
+                return true; //Table exists
+            else
+                return false;
+        }
+    }
+
+    private List<String> tableExists(List<String> tables, File clibFile) throws SQLException
+    {
+        try (Connection conn = ConnectionSource.getConnection(clibFile.getAbsolutePath()))
+        {
+            DatabaseMetaData md = conn.getMetaData();
+            for (Iterator<String> i = tables.iterator(); i.hasNext(); )
+            {
+                String tableName = i.next();
+                ResultSet rs = md.getTables(null, null, tableName, null);
```

Comment:
These ResultSets should be closed as well via a try-with-resource syntax. Your push overwrote my example that I pushed for columnExists(). See https://github.com/LabKey/targetedms/pull/325/commits/53c56b78ea54262089151cc9c4b732b748e8c828

Reply:
Yup, sorry for that.